test/unit/controllers/declaration/ExporterDetailsControllerSpec.scala

```diff
@@ -0,0 +1,107 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package unit.controllers.declaration
+
+import controllers.declaration.ExporterDetailsController
+import forms.common.Address
+import forms.declaration.ExporterDetails
+import org.mockito.{ArgumentCaptor, Mockito}
+import org.mockito.ArgumentMatchers.any
+import org.scalatest.OptionValues
+import play.api.data.Form
+import play.api.libs.json.Json
+import play.twirl.api.HtmlFormat
+import unit.base.ControllerSpec
+import views.html.declaration.exporter_details
+import play.api.test.Helpers._
+
+import scala.concurrent.ExecutionContext
+
+class ExporterDetailsControllerSpec extends ControllerSpec with OptionValues {
+
+  import forms.declaration.ExporterDetailsSpec._
+
+  val exporter_details = mock[exporter_details]
+
+  val controller = new ExporterDetailsController(
+    mockAuthAction,
+    mockJourneyAction,
+    mockExportsCacheService,
+    stubMessagesControllerComponents(),
+    exporter_details
+  )(ExecutionContext.global)
+
+  def templateArgument: Form[ExporterDetails] = {
+    val captor = ArgumentCaptor.forClass(classOf[Form[ExporterDetails]])
+    Mockito.verify(exporter_details).apply(captor.capture())(any(), any())
+    captor.getValue
+  }
+
+  override protected def beforeEach(): Unit = {
+    super.beforeEach()
+    authorizedUser()
+    Mockito.when(exporter_details.apply(any())(any(), any())).thenReturn(HtmlFormat.empty)
+  }
+
+  override protected def afterEach(): Unit = {
+    Mockito.reset(exporter_details)
+    super.afterEach()
+  }
+
+  "Exporter Details Controller" should {
+    "return 200 OK" when {
+      "details are empty" in {
+        val declaration = aDeclaration()
+        withNewCaching(declaration)
+        val response = controller.displayForm().apply(getRequest(declaration))
+        status(response) mustBe OK
+      }
+      "details are filled" in {
+        val declaration = aDeclaration(
+          withExporterDetails(
+            eori = Some("99980"),
+            address = Some(Address("CaptainAmerica", "Test Street", "Leeds", "LS18BN", "Portugal"))
+          )
+        )
+        withNewCaching(declaration)
+        val response = controller.displayForm().apply(getRequest(declaration))
+        status(response) mustBe OK
+        val details = templateArgument.value.value.details
+        details.eori mustBe defined
+        details.address mustBe defined
+      }
+    }
+    "return 400 bad request" when {
+      "form contains erros" in {
+        val declaration = aDeclaration()
+        withNewCaching(declaration)
+        val response = controller.saveAddress().apply(postRequest(Json.obj(), declaration))
+        status(response) mustBe BAD_REQUEST
```

Comment:
Should we assert on what arguments the view was built with?

Reply:
In this example empty request ware made - we can assert that view ware called